src/VisualStudio/Core/Def/DocumentOutline/DocumentOutlineControl.xaml.cs

```diff
@@ -203,25 +186,52 @@ int IVsCodeWindowEvents.OnCloseView(IVsTextView pView)
         }
 
         private void OnEventSourceChanged(object sender, TaggerEventArgs e)
-            => EnqueueComputeDataModelTask();
+            => EnqueuGetSettingsTask();
 
         /// <summary>
         /// On caret position change, highlight the corresponding symbol node in the window and update the view.
         /// </summary>
         private void Caret_PositionChanged(object sender, CaretPositionChangedEventArgs e)
         {
             if (!e.NewPosition.Equals(e.OldPosition))
-                EnqueueHighlightExpandAndPresentItemsTask(ExpansionOption.CurrentExpansion);
+                EnqueueUpdateUITask(ExpansionOption.CurrentExpansion, TryGetCurrentCaretPoint(e.NewPosition));
+        }
+
+        private SnapshotPoint? TryGetCurrentCaretPoint(CaretPosition? newPosition = null)
+        {
+            var activeTextView = GetLastActiveIWpfTextView();
+            if (activeTextView is null)
+            {
+                return null;
+            }
+
+            var textBuffer = activeTextView.TextBuffer;
```

Comment:
using this text-buffer seems iffy.  it likely wouldn't work in projection scenarios right?  we likely need the text buffer for the actual roslyn document.